newIDE/app/src/UI/Menu/ElementWithMenu.js

```diff
@@ -5,45 +5,78 @@ import ContextMenu from './ContextMenu';
 
 type Props = {|
   element: React$Element<any>,
-  buildMenuTemplate: () => Array<any>,
+  onClick?: () => void,
+  buildMenuTemplate?: () => Array<any>,
+  buildMenuTemplateRight?: () => Array<any>,
 |};
 
 type State = {||};
 
 /**
  * Wrap an element and display a menu when `onClick` prop is called on the element.
  */
+
 export default class ElementWithMenu extends React.Component<Props, State> {
   _contextMenu: ?ContextMenu;
+  _contextMenuR: ?ContextMenu;
```

Comment:
I'm thinking of a slightly different implementation. Keep a single context menu (`_contextMenu`) *but* add a prop to `ElementWithMenu` that, if set to true, will only show the menu when you do a right click.

This will avoid duplicating code for open/openRight and also this avoid the temptation of doing a button with a context menu AND a right click context menu.
See my other comment to do this :)